app/code/Magento/Review/Block/Adminhtml/Main.php

```diff
@@ -37,26 +43,37 @@ class Main extends \Magento\Backend\Block\Widget\Grid\Container
      */
     protected $_customerViewHelper;
 
+    /**
+     * Product Collection
+     *
+     * @var \Magento\Catalog\Model\ResourceModel\Product\Collection
+     */
+    protected $productCollection;
+
     /**
      * @param \Magento\Backend\Block\Widget\Context $context
      * @param \Magento\Customer\Api\CustomerRepositoryInterface $customerRepository
      * @param \Magento\Catalog\Model\ProductFactory $productFactory
      * @param \Magento\Framework\Registry $registry
      * @param \Magento\Customer\Helper\View $customerViewHelper
      * @param array $data
+     * @param \Magento\Catalog\Model\ResourceModel\Product\Collection $productCollection
      */
     public function __construct(
         \Magento\Backend\Block\Widget\Context $context,
         \Magento\Customer\Api\CustomerRepositoryInterface $customerRepository,
         \Magento\Catalog\Model\ProductFactory $productFactory,
         \Magento\Framework\Registry $registry,
         \Magento\Customer\Helper\View $customerViewHelper,
-        array $data = []
+        array $data = [],
+        \Magento\Catalog\Model\ResourceModel\Product\Collection $productCollection = null
```

Comment:
Don't inject collections. Inject factories and prepare this collection as you need. It will be more obvious for developers what this collection contains. And of course the current way you can't be sure you will get collection that you need here all the time.